src/main/java/edu/umich/tl/analytics/CanvasCourseReportFilter.java

```diff
@@ -0,0 +1,197 @@
+				M_log.error("Failed to load application properties from canvasReport.properties");
+			}
+			
+		}else {
+			M_log.error("File path for (canvasReport.properties) is not provided");
+		}
+	}
+
+	public void doFilter(ServletRequest sReq, ServletResponse sRes, FilterChain chain)
+			throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) sReq;
+		HttpServletResponse useResponse=(HttpServletResponse)sRes;
+		if(!checkForAuthorization(useRequest)) {
+			useResponse.sendError(403);
+			return;
+		}
+		useRequest.setAttribute("canvasURL",canvasURL);
+		chain.doFilter(useRequest, useResponse);
+		
+	}
+	private boolean checkForAuthorization(HttpServletRequest request) {
+		 M_log.debug("checkLdapForAuthorization(): called");
+		 String remoteUser = request.getRemoteUser();
+		 String testUser = request.getParameter(TEST_USER);
+		 boolean isAuthorized = false;
+		 String user=null;
+
+		 String testUserInSession = (String)request.getSession().getAttribute(TEST_USER);
+		 String sessionId = request.getSession().getId();
+
+		 if ( isTestUrlEnabled && testUser != null ) { 
+			 user=testUser;
+			 request.getSession().setAttribute(TEST_USER, testUser);
+		 }
+		 else if ( isTestUrlEnabled && testUserInSession != null ){
+			 user=testUserInSession;
+		 } 
+		 if  ( !isAuthorized && remoteUser != null ) {
+			 user=remoteUser;
+			 M_log.info(String.format("The session id \"%s\" of the Person with uniqname  \"%s\" issuing the request" ,sessionId,remoteUser));
+		 }
+		 isAuthorized=ldapAuthorizationVerification(user); 
+		 return isAuthorized;
+	}
+	
+	private boolean ldapAuthorizationVerification(String user)  {
+		M_log.debug("ldapAuthorizationVerification(): called");
+		boolean isAuthorized = false;
+		DirContext dirContext=null;
+		NamingEnumeration listOfPeopleInAuthGroup=null;
+		NamingEnumeration allSearchResultAttributes=null;
+		NamingEnumeration simpleListOfPeople=null;
+		Hashtable<String,String> env = new Hashtable<String, String>();
+		if(!ReportUtilities.isEmpty(providerURL) && !ReportUtilities.isEmpty(mcommunityGroup)) {
+		env.put(Context.INITIAL_CONTEXT_FACTORY, LDAP_CTX_FACTORY);
+		env.put(Context.PROVIDER_URL, providerURL);
+		}else {
+			M_log.error(" [ldap.server.url] or [mcomm.group] properties are not set, review the canvasReport.properties file");
+			return isAuthorized;
+		}
+		try {
+			dirContext = new InitialDirContext(env);
+			String[] attrIDs = {"member"};
+			SearchControls searchControls = new SearchControls();
+			searchControls.setReturningAttributes(attrIDs);
+			searchControls.setReturningObjFlag(true);
+			searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+			String searchBase = OU_GROUPS;
+			String filter = "(&(cn=" + mcommunityGroup + ") (objectclass=rfc822MailGroup))";
+			listOfPeopleInAuthGroup = dirContext.search(searchBase, filter, searchControls);
+			String positiveMatch = "uid=" + user + ",";
+			outerloop:
```

Comment:
It would be good to add a comment on why it makes sense to have nested loops for this code rather than having a smaller loop calling out to methods.